src/engine/pieces/piece.ts

```diff
@@ -17,4 +17,15 @@ export default class Piece {
         const currentSquare = board.findPiece(this);
         board.movePiece(currentSquare, newSquare);
     }
+
+    public checkMoveConflicts(board: Board, possibleMoves: Square[], currentSquare: Square){
+        const occupiedSquares: Square[] = []
+        for (let square of possibleMoves){
```

Comment:
This can be done using some functional programming 😄 
```
occupiedSquares = possibleMoves.filter((square) => board.getPiece(square) != null)
unoccupiedSquares = possibleMoves.filter((square) => board.getPiece(square) == null)
```

Let me know if you'd like an explanation on how this works!